app/services/data_exports/create_service.rb

```diff
@@ -41,18 +41,15 @@ def validate_params
     # Find the project_ids for each sample, and search/validate the unique set of ids to ensure user has authorization
     # to export the chosen samples' data
     def validate_sample_ids
-      project_ids = []
-
-      params['export_parameters']['ids'].each do |sample_id|
-        sample = Sample.find_by(id: sample_id)
-        raise DataExportCreateError, I18n.t('services.data_exports.create.invalid_sample_id') if sample.nil?
-
-        project_id = sample.project_id
-        project_ids << project_id unless project_ids.include?(project_id)
+      samples = Sample.where(id: params['export_parameters']['ids'])
```

Comment:
**nitpick (blocking):** at the end of this `where` you could use `.select(:id, :project_id)` as those are the only 2 attributes that you are using.

Reply:
I get a crash when using `samples = Sample.where(id: params['export_parameters']['ids']).select(:id, :project_id)`. It seems happy when using either or. Am I missing something having the two arguments?

Reply:
Made the changes as we discussed in fe2ede75f9d4240f8b71eefb3d4dce58624b27ed